ts/common/sheet_helpers.ts

```diff
@@ -1388,10 +1388,23 @@ function extract(content: string): string {
 /**
  * Generates an email body given a list of possibly anomalous values.
  */
-function emailAlertBody(name: string, values: Value[]) {
-  const header = `----------\n${name}:\n----------\n`;
+export function emailAlertBody(name: string, values: Value[]) {
+  const rule = `----------\n${name}:\n----------\n\n`;
+  const headers = new Set<string>();
+  const body: string[][] = [];
+  for (const value of values) {
+    for (const field of Object.keys(value.fields)) {
+      // adding all headers except value which goes at the end.
+      headers.add(field);
+    }
+    body.push(Object.values(value.fields).concat([value.value]));
+  }
+  headers.add('Value');
   const anomalyList =
-    header + values.map((value) => `- ${value.value}`).join('\n');
+    rule +
```

Comment:
nit: tsstyle [says](https://google.github.io/styleguide/tsguide.html#:~:text=Use%20template%20literals%20(delimited%20with%20%60)%20over%20complex%20string%20concatenation) to use string templates instead of complex concatenation. I prefer it here mostly to convey that, yes, this is a string, and not something else.